### PR TITLE
RHPAM-1550 - User is unable invoke DMN model from business process

### PR DESCRIPTION
**Pull request by Josephblt**
https://issues.jboss.org/browse/RHPAM-1550

Assignment Variable Name Field now accepts "Space" as a valid character.

### CODE REVIEW

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Hello @jomarko, @romartin. If you guys could take a look at this, that be great. Thanks.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/util/StringUtilsTest.java [hunk 1]
```diff
@@ -74,4 +76,26 @@ public void testCreateDataTypeDisplayName() throws Exception {
...
+        String test1 = "123Test";
+        assertTrue(test1.matches(StringUtils.ALPHA_NUM_REGEXP));
+
+        String test2 = " Test";
+        assertFalse(test2.matches(StringUtils.ALPHA_NUM_REGEXP));
+
+        String test3 = "Test#";
+        assertFalse(test3.matches(StringUtils.ALPHA_NUM_REGEXP));
+
+        String test4 = "123Test";
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/util/StringUtilsTest.java]: Don't we miss the space in this input?

**Comment by jomarko** [comment]
@Josephblt Thank you for the PR, the changes enables me to invoke the DMN model with whitespaces in variable names. However I realized the dmn/bpmn variable naming difference is not only in whitespaces. DMN allows also characters like **?** etc. The good thing is there is easy way to check DMN variable name, see `org.kie.dmn.feel.parser.feel11.FEELParser.isVariableNameValid()`, the bad thing is I am not sure if we could/should add a drools dependency into a stunner module.

@romartin would be adding such dependency fine?


@krisv @etirelli Hi, question for you as product owners, do we want to fully support bpmn<->dmn interoperability, or is it fine if bpmn supports just dmn names with whitespaces, it means are we fine if bpmn doesn't support dmn names like **is person old?**

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/util/StringUtilsTest.java (hunk 1 above)
Comment (Josephblt) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/util/StringUtilsTest.java]: Actually no. What I wanted to test in this line is just a invalid character other than a white space. Could add "123" to match the others inputs.

Comment (jomarko) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/forms/util/StringUtilsTest.java]: fine, then

**Comment by etirelli** [comment]
@jomarko no, we need to support all special characters that DMN supports:

The list of characters is: ```+ - / * ? ' .```

We don't need to add the FEEL parser dependency just to check these characters. Just check for them in a specific function here.

**Comment by jomarko** [comment]
@etirelli thank you for answer, it answered the more important part. It is clear now PR need to be updated. However personally I do not understand why not to reuse logic of `FEELParser` and `reimplement` it form scratch here.

**Comment by etirelli** [comment]
@jomarko not saying the FEELParser shouldn't be used, just that it might not be worth adding a new dependency just because of this simple rule about these few characters that can be used. 

If the dependency is to be added for additional reasons, then absolutely use the FEELParser.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
All done now @jomarko. Ready to test.  Thanks.

**Comment by romartin** [comment]
@jomarko I'm agree with @etirelli about not adding the drools dependency, if possible, because it'll end up on several more transitive dependencies .. and also probably the main reason for not doing that, if I'm correct, is that this check is being done on the widget, so on client side, right? Not really sure if FEELParser is available on client side, at least I think the `drools` modules is not...

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
ping @jomarko plz - this one is urgent, could you please complete your review? thanks!

**Comment by romartin** [comment]
Could you please @hasys  or  @LuboTerifaj check this commit doesn't break BPMN or CM stuff as well? We're really close to the candidate release and having small margin to react to issues introduced now. Thanks!!

**Comment by hasys** [comment]
Hi @romartin, @jomarko, already tested BPMN Designer, and CM Designer doesn't have Ruleflow task so it is not affected. Approval from @jomarko should be enough. Thank you @jomarko!

**Comment by romartin** [comment]
@hasys @LuboTerifaj @Josephblt @jomarko builds are never ending....

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Had to push meaningless changes to stop the build. No need to retest. This is ready to merge.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Helo @romartin. Failed tests on the full downstream build are not related to this PR. Again, no need to retest. Ready to merge.

**Comment by Josephblt** [comment]
Hello @jomarko. Changing allowed characters in the BPMN assignment popup is just a matter of changing the regex pattern. If you guys deem it necessary just tell me, it is a quick change.

**Pull request merged**